xmppserver/src/main/java/org/jivesoftware/openfire/muc/HistoryStrategy.java

```diff
@@ -55,34 +57,47 @@
     /**
      * List containing the history of messages.
      */
+    // TODO it is likely that a lot of serialization (in a cluster) can be prevented by replacing this queue with a clustered cache.
     private ConcurrentLinkedQueue<Message> history = new ConcurrentLinkedQueue<>();
+
     /**
      * Default max number.
      */
     private static final int DEFAULT_MAX_NUMBER = 25;
+
     /**
      * The maximum number of chat history messages stored for the room.
      */
     private int maxNumber;
+
     /**
      * The parent history used for default settings, or null if no parent
      * (chat server defaults).
      */
     private HistoryStrategy parent;
+
     /**
      * Track the latest room subject change or null if none exists yet.
      */
     private Message roomSubject = null;
+
     /**
      * The string prefix to be used on the context property names
      * (do not include trailing dot).
      */
     private String contextPrefix = null;
+
     /**
      * The subdomain of the service the properties are set on.
      */
     private String contextSubdomain = null;
 
+    /**
+     * This constructor is provided to comply with the Externalizable interface contract. It should not be used directly.
```

Comment:
This seems like a bit of a smell. Should a strategy really be Externalizable? Should there be two classes, one Stragy, one to externalise the state?